docs/running-tests/from-local-system.md

```diff
@@ -4,69 +4,86 @@ The tests are designed to be run from your local computer.
 
 ## System Setup
 
-The test environment requires [Python 2.7+](http://www.python.org/downloads)
-(but not Python 3.x).
+Running the tests requires `python`, `pip` and `virtualenv`, as well as updating
+the system `hosts` file.
 
-On Windows, be sure to add the Python directory (`c:\python2x`, by default) to
-your `%Path%` [Environment Variable](http://www.computerhope.com/issues/ch000549.htm),
-and read the [Windows Notes](#windows-notes) section below.
+Note that Python 2.7 is required. Using Python 3 is not supported.
 
-<!--
-  There does not appear to be a cross-platform means of installing `pip`.
-  https://github.com/web-platform-tests/wpt/pull/16670
--->
+The required setup is different depending on your operating system.
 
-Install `pip`. On many systems, this can be achieved with the command `python
--m ensurepip`. If this is not possible, use your system's package manager to
-install the `python-pip` package.
+### Linux Setup
 
-Next, install `virtualenv` using the following command:
+If not already present, use the system package manager to install `python`,
+`pip` and `virtualenv`.
+
+On Debian or Ubuntu:
 
 ```bash
-pip install virtualenv
+sudo apt-get install python python-pip virtualenv
 ```
 
-To get the tests running, you need to set up the test domains in your
-[`hosts` file](http://en.wikipedia.org/wiki/Hosts_%28file%29%23Location_in_the_file_system).
-
-The necessary content can be generated with `./wpt make-hosts-file`; on
-Windows, you will need to preceed the prior command with `python` or
-the path to the Python binary (`python wpt make-hosts-file`).
-
-For example, on most UNIX-like systems, you can setup the hosts file with:
+On Fedora or Red Hat:
 
 ```bash
-./wpt make-hosts-file | sudo tee -a /etc/hosts
+sudo dnf install python2 python2-pip python2-virtualenv
```

Comment:
@clopez I think that python2-virtualenv should work on Fedora stable, but can you confirm?

Reply:
I have tested this on a docker image of fedora31 and unfortunately that is not enough.
Issues:
  * `/usr/bin/python` doesn't exist anymore on fedora. Possible workarounds:
    * Tell the user to create the symlink with: ```sudo ln -s /usr/bin/python2 /usr/bin/python```
    * We update the shebangs on all the python scripts in the wpt repository to use `/usr/bin/python2` or `/usr/bin/python2.7` instead of `/usr/bin/python`. I think is maybe better `/usr/bin/python2.7` because IIRC some versions of Mac (previous to Catalina) didn't had a ```python2``` symlink/binary, but they had a ```python2.7``` one
  * The package python2-virtualenv doesn't exist anymore.  Possible workaround:
     * Tell the user to install it via pip with ```pip2 install virtualenv```

So the instructions would be:

```
sudo dnf install python2 python2-pip
sudo ln -s /usr/bin/python2 /usr/bin/python
pip2 install virtualenv
```

However, I don't like the idea of telling the user to create a symlink in `/usr/bin`. Maybe we should consider changing the shebangs on our python scripts. I tested to change it only on the main one (`./wpt`) and it seems to work:
```diff
diff --git a/wpt b/wpt
index a24bc83a91..bc9924a9e0 100755
--- a/wpt
+++ b/wpt
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 
 if __name__ == "__main__":
     from tools.wpt import wpt
```

BTW, is there any plan to migrate the codebase to python3?

Reply:
https://github.com/web-platform-tests/wpt/issues/8820 is the tracking issue for Python 3.

Since work is needed to make `./wpt run` work on Fedora and it's not just documentation, I'll revert these instructions and file an issue linking to your very helpful comment here.

Reply:
Filed https://github.com/web-platform-tests/wpt/issues/20035.

Reply:
We should fix this by fixing the wpt front-end. Carlos' patch looks right.

Reply:
SGTM. There are lots of occurrences of `/usr/bin/python` and `/usr/bin/env python` without a trailing number in the repo, so doing something across the board would be good. Otherwise `./wpt run` might work, but lots of other things won't.